hal/stm32u5.c

```diff
@@ -323,24 +386,30 @@ int RAMFUNCTION hal_flash_write(uint32_t address, const uint8_t *data, int len)
 #endif
 
     while (i < len) {
-        qword[0] = src[i >> 2];
-        qword[1] = src[(i >> 2) + 1];
-        qword[2] = src[(i >> 2) + 2];
-        qword[3] = src[(i >> 2) + 3];
+        /* load the start of the word */
+        qword = *(uint32_t*)(((uint32_t)dst) -
+            (((uint32_t)dst) % 4));
+
+        do {
+            /* shift up by the number of bytes past word align */
+            offset = (((uint32_t)dst + i) % 4);
+            qword |= ((uint32_t)data[i] << offset);
+            i++;
```

Comment:
no I cast to uint8_t so I can write byte by byte instead of 4 bytes at a time. this is so we can write the single status byte without overwriting the surrounding bytes

Reply:
also no I haven't tested, my stm32u5 dev board gets here on saturday